lib/eslint/eslint-helpers.js

```diff
@@ -675,7 +675,6 @@ function processOptions({
     overrideConfig = null,
     overrideConfigFile = null,
     plugins = {},
-    reportUnusedDisableDirectives = null, // ← should be null by default because if it's a string then it overrides the 'reportUnusedDisableDirectives' setting in config files. And we cannot use `overrideConfig.reportUnusedDisableDirectives` instead because we cannot configure the `error` severity with that.
```

Comment:
Since this is an API change when compared to previous APIs that had this option, it would be good to show a descriptive message if someone tries to pass this option. Like, for example, we're doing with the `globals` options:

https://github.com/eslint/eslint/blob/e0cb960e45961a36845b075a2986a1fa6cd6b8aa/lib/eslint/eslint-helpers.js#L702-L704

```js
if (unknownOptionKeys.includes("reportUnusedDisableDirectives")) {
    errors.push("'reportUnusedDisableDirectives' has been removed. Please use the 'overrideConfig.linterOptions.reportUnusedDisableDirectives' option instead.");
}
```